<launch>
    <arg name="slave_1_name" value="PSM1_DUMMY"/>
    <arg name="slave_2_name" value="PSM2_DUMMY"/>
    <arg name="master_1_name" value="MTMR"/>
    <arg name="master_2_name" value="MTML"/>

    <!-- parameters used by some nodes regarding the transformations among the cams, arms and task-space -->
    <group ns="calibrations">
        <rosparam command="load" file="$(find atar)/launch/params_calibrations_test_vr.yaml" />
    </group>

    <!-- Node GUI and recording the data-->
    <node pkg="atar" type="gui" name="gui" output="screen">
        <!--records data at the frequency of task_state topic-->
        <param name="number_of_arms" value="2"/>
        <param name="slave_1_name" value="$(arg slave_1_name)"/>
        <param name="master_1_name" value="$(arg master_1_name)"/>
        <param name="slave_2_name" value="$(arg slave_2_name)"/>
        <param name="master_2_name" value="$(arg master_2_name)"/>
    </node>

    <!-- This node is now supposed to generate the ac geometry, control task states, and overlay graphics-->
    <node pkg="atar" type="ar_core" name="ar_core" output="screen">

        <!-- About stl_files_dir:
        The directory of the stl files -->
        <param name= "mesh_directory" value= "$(find atar)/resources/mesh/" />

        <!--Activate or deactivate haptic guidance -->
        <param name= "publish_overlayed_images" value= "false" />

        <!--add shadows to the graphics. Works on dedicated GPUs -->
        <!--NOTE: SHADOWS DON'T WORK IF THERE ARE MORE THAN ONE WINDOW.MULTIPLE VIEWS IN ONE WINDOW IS FINE-->
        <param name= "with_shadows" value= "true" />

        <!--If the generated images are to be published on
        ros, to help alleviate the considerable bottleneck of grabbing the
        images from the gpu, activate this flag so that the rendering is done
         off screen and not shown in opengl windows.-->
        <param name= "offScreen_rendering" value= "false" />

    </node>

    <node pkg="atar" type="teleop_dummy_dvrk" name="teleop_dummy_dvrk"
          output="screen">
        <param name="slave_1_name" value="$(arg slave_1_name)"/>
        <param name="master_1_name" value="$(arg master_1_name)"/>
        <param name="slave_2_name" value="$(arg slave_2_name)"/>
        <param name="master_2_name" value="$(arg master_2_name)"/>
        <param name="scaling" value="0.5"/>

    </node>

    <!-- The dvrk_active_constraint_enforcement node generates wrench based on the received desired pose and
        current pose of each master/slave couple.
	The node generates wrench when receives desired pose, so the loop rate depends on the frequency of the
	desired pose topic. The sampling_frequency param is for filtering.
    -->
    <node pkg="active_constraints" type="dvrk_active_constraint_enforcement" name="dvrk_ac_enforce" output="screen">
        <param name="number_of_arms" value="2"/>
        <param name="slave_1_name" value="$(arg slave_1_name)"/>
        <param name="slave_2_name" value="$(arg slave_2_name)"/>
        <param name="master_1_name" value="$(arg master_1_name)"/>
        <param name="master_2_name" value="$(arg master_2_name)"/>
        <!--the ac params will be overwritten if new message arrives-->
        <param name="max_force" value="3"/>
        <param name="max_torque" value="0.03"/><!-- 0.03 -->
        <param name="linear_elastic_coeff" value="1000"/>
        <param name="linear_damping_coeff" value="10"/>
        <param name="angular_elastic_coeff" value="0.04"/>
        <param name="angular_damping_coeff" value="0.002"/>
        <param name="sampling_frequency" value="500"/>
    </node>

</launch>
